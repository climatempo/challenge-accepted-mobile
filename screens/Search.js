import React, { useState, useEffect } from 'react';
import { View, FlatList, Text, ScrollView } from 'react-native';
import { Button, Card, TextInput } from 'react-native-paper';
import axios from 'axios'
import { TOKENAPI } from '../utils'
import AsyncStorage from '@react-native-async-storage/async-storage'
import styles from '../public/searchStyle'
const Search = ({ navigation }) => {
    const [cidade, setCidade] = useState('')
    const [cidades, setCidades] = useState([])
    const [loading, setLoading] = useState(false)
    const [cidPesquisadas, setcidPesquisadas] = useState([])

    useEffect(() => {

        const hasPastCity = async () => {

            let idCidade = await AsyncStorage.getItem('idCidade')
            if (idCidade !== null) {
                idCidade = JSON.parse(idCidade)
                navigation.navigate('Cidade', { idLocale: idCidade })
            }
        }
        hasHistory();
        hasPastCity();
    }, [])

    const hasHistory = async () => {

        let historys = await AsyncStorage.getItem('history');
        if (historys !== null) {
            setcidPesquisadas(JSON.parse(historys))
            console.log('History Storage:' + historys)
        }
    }


    const saveHistory = async () => {
        setcidPesquisadas(cidPesquisadas => [...cidPesquisadas, cidade]);
        console.log('Cid Pesquisadas aqui: ' + cidPesquisadas)
        await AsyncStorage.setItem('history', JSON.stringify(cidPesquisadas));
        let historys = await AsyncStorage.getItem('history');
        let arr = []
        if (historys !== null) {
            arr.push(JSON.parse(historys))
            console.log('After Save: ' + arr)
        }
    }

    const fetchLocales = async () => {
        setLoading(true)
        await saveHistory();
        const { data } = await axios.get(`http://apiadvisor.climatempo.com.br/api/v1/locale/city?name=${cidade}&token=${TOKENAPI}`)

        if (data) {
            setCidades(data)
            setLoading(false)
        }
    }

    const History = () => {
        return (
            <>
                {
                    cidPesquisadas.map(item => (
                        <View key={item} style={styles.row}>
                            <Text style={styles.searchItem}>{item}</Text>
                            <Button color={'white'} style={styles.buttonDelete} onPress={() => removeFilter(item)}>Deletar</Button>
                        </View>
                    )
                    )
                }
            </>
        )
    }

    const removeFilter = (strFilter) => {
        var array = [...cidPesquisadas]
        var index = cidPesquisadas.findIndex(x => x == strFilter)
        if (index !== -1) {
            array.splice(index, 1)
            setcidPesquisadas(array)
        }
    }

    const removeFilterAll = async () => {
        await AsyncStorage.removeItem('history')
        setcidPesquisadas([])
        alert('Histórico Apagado!')
    }

    const goToCity = async (item) => {
        setCidade(item.name)
        navigation.navigate('Cidade', { idLocale: item.id })
        console.log('ID: ' + item.id)
    }


    const renderItem = ({ item }) => {
        return (
            <Card style={styles.card} onPress={() => goToCity(item)}>
                <Text style={styles.font20}>{item.name} - {item.state}</Text>
            </Card>
        )
    }


    return (
        <View style={styles.container}>
            <TextInput
                label="Nome da cidade"
                value={cidade}
                onChangeText={text => setCidade(text)} />
            <Button mode="contained" style={styles.buttonPesq} color="#932ad4" onPress={() => fetchLocales()}>
                Pesquisar
            </Button>

            {loading ? <Text>Carregando...</Text> :

                <View>
                    <Text style={styles.font20}>Histórico de pesquisa:</Text>
                    <ScrollView>
                        <History />
                        <Button color={'white'} style={styles.buttonDeleteAll} onPress={() => removeFilterAll()}>APAGAR TODOS</Button>

                    </ScrollView>

                    <Text style={styles.font20}>Resultados encontrados:</Text>
                    <FlatList
                        data={cidades}
                        renderItem={renderItem}
                        keyExtractor={item => item.id.toString()}
                    />
                </View>
            }
        </View>
    );
};


export default Search;
