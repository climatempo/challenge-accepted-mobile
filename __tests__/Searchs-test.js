import 'react-native';
import React from 'react';
import Search from '../screens/Search';
import renderer from 'react-test-renderer'

test('Should render Search component', () => {
    const snap = renderer.create(
        <Search />
    ).toJSON();

    expect(snap).toMatchSnapshot();
})

