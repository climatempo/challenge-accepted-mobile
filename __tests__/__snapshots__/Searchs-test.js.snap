// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Object {
          "minHeight": 64,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Nome da cidade
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Nome da cidade
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#932ad4",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 10,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Pesquisar
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Hist√≥rico de pesquisa:
    </Text>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#a83232",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "margin": 5,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "white",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                APAGAR TODOS
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Resultados encontrados:
    </Text>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;

exports[`Should render Search component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(231, 231, 231)",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.26)",
          "bottom": 0,
          "height": 2,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "transform": Array [
            Object {
              "scaleY": 0.5,
            },
          ],
        }
      }
    />
    <View
      style={
        Object {
          "minHeight": 64,
          "paddingBottom": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <Text
          numberOfLines={1}
          onLayout={[Function]}
          style={
            Object {
              "color": "#6200ee",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Nome da cidade
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "left": 0,
              "opacity": 0,
              "paddingLeft": 12,
              "paddingRight": 12,
              "position": "absolute",
              "textAlign": "left",
              "top": 34,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
                Object {
                  "scale": 1,
                },
              ],
              "writingDirection": "ltr",
            }
          }
        >
          Nome da cidade
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder=""
        placeholderTextColor="rgba(0, 0, 0, 0.54)"
        rejectResponderTermination={true}
        selectionColor="#6200ee"
        style={
          Array [
            Object {
              "flexGrow": 1,
              "margin": 0,
              "zIndex": 1,
            },
            Object {
              "paddingLeft": 12,
              "paddingRight": 12,
            },
            Object {
              "height": 64,
            },
            Object {
              "paddingBottom": 4,
              "paddingTop": 24,
            },
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            Array [
              Object {},
            ],
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#932ad4",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "margin": 10,
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Pesquisar
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Hist√≥rico de pesquisa:
    </Text>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#a83232",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "margin": 5,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "white",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                APAGAR TODOS
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Resultados encontrados:
    </Text>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
